### PR TITLE
feat: Supports quickly switching accounts with a swipe-up gesture on …

### PR DESCRIPTION
**Pull request by dploeger**
…the tab bar on iOS

Like I said on Mastodon, this is my first dabble in SwiftUI and Swift, so here be dragons (probably).

It implements switching the current account by simply swiping up on the tab bar. I'm open to suggestions.

### CODE REVIEW

File: IceCubesApp/App/IceCubesApp-release.entitlements [hunk 1]
```diff
@@ -2,22 +2,10 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>aps-environment</key>
```
Comment (Dimillian) [inline IceCubesApp/App/IceCubesApp-release.entitlements:5]: This should probably not be commited

File: IceCubesApp/App/IceCubesApp.entitlements [hunk 2]
```diff
@@ -2,22 +2,10 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>aps-environment</key>
```
Comment (Dimillian) [inline IceCubesApp/App/IceCubesApp.entitlements:5]: Same

File: IceCubesApp/App/IceCubesApp-release.entitlements (hunk 1 above)
Comment (dploeger) [reply IceCubesApp/App/IceCubesApp-release.entitlements:5]: Arrgggh. And here I am trusting XCode with Git commands. Sorry.

**Comment by Dimillian** [comment]
It's working! Now I wonder why not a long tab to bring the account selector too?

**Comment by dploeger** [comment]
I've removed the long tap. Should I re-add it?

**Comment by Dimillian** [comment]
I'm wondering if the swipe should swap the account or present the account selector sheet. I think maybe present the account selector sheet?

**Comment by dploeger** [comment]
Maybe we could do both. The swipe is (or "feels") like a quick gesture, used for doing stuff swiftly, while the long press is for something more slower.

So maybe: long press for account selector sheet and swipe for switching the account?

**Comment by Dimillian** [comment]
Sound good to test

**Comment by dploeger** [comment]
Could you please point me to the new code where you open the compose sheet in a tab? I'm not really sure how to trigger opening a sheet in SwiftUi.

**Comment by Dimillian** [comment]
Hmm it won't be easy as the app account selector is not globally available. I would need to refactor a few stuff. For now it's in `AppAccountsSelectorView` and there is no way to trigger it easily. It's bundled with the rounded avatar button basically.

**Comment by dploeger** [comment]
Ah account. Let me see, if I can manage the refactoring...

**Comment by dploeger** [comment]
Done. I've also rebased and saw how you implemented the sheet of the New button. Maybe the accountsSelectorView can be implemented the same way, but please just take a look.

**Comment by Dimillian** [comment]
I'm testing it a bit, and I find that a tad confusing. Not sure this is the right direction :/

**Comment by dploeger** [comment]
Yeah, I'm with you on that. The code needs to look more beautiful.

**Comment by Dimillian** [comment]
> Yeah, I'm with you on that. The code needs to look more beautiful.

I was thinking about the UX, too. It's hard to discover. The TabView from SwiftUI is not really super extensible.

**Comment by dploeger** [comment]
That's true, but I think the flick gesture to switch accounts is a nice UX, thought it has to be documented because, yes, it's not obvious. But in day-to-day usage it's pretty nice. I could live without the long-press gesture tbh.

**Pull request closed without merging**